### ISSUE TITLE
GetEDSMDistances 'System.OutOfMemoryException'

### ISSUE THREAD

**Issue by Aethir** [EDDiscovery/EDDiscovery#65]
I so desperately want this to work... 
Have had a few dramas with getting EDDisco to play nice. I enabled VerboseLogging in the game config, and since then I'm getting the abovementioned error. In full, this is what EDDisco shows me:

EliteDangerous is not running .
Checking for new EDDiscovery data
Checking for new EDDiscovery maps
C:\Users\Krang\AppData\Local\EDDiscovery\tgcsystems.json is up to date.
1576 new systems from EDSM.
Reading travelhistory Downloading mirrored EDSM distance data. (Might take some time)
Get systems from EDDB. 
Loading completed!
OK.
Get stations from EDDB. OK.
Add new EDDB data to database.
EDDB update done.
Adding mirrored EDSM distance data.
GetEDSMDistances exception:Exception of type 'System.OutOfMemoryException' was thrown.
   at Newtonsoft.Json.Linq.JContainer.ReadContentFrom(JsonReader r)
   at Newtonsoft.Json.Linq.JContainer.ReadTokenFrom(JsonReader reader)
   at Newtonsoft.Json.Linq.JArray.Load(JsonReader reader)
   at Newtonsoft.Json.Linq.JArray.Parse(String json)
   at EDDiscovery.DB.DistanceClass.ParseEDSM(String json, String& date)
   at EDDiscovery.EDDiscoveryForm.GetEDSMDistances()

I'm using Win 7 Pro on a system with 16Gb RAM, of which 78% is currently free...

I'd really like to get this program up and running before I head anywhere outside the Bubble... so any assistance would be awesome :)
Thanks in advance

**Comment by finwen** [comment]
Short term solution is to uncheck  Get Distances  in the setings tab.

The problem is that the json parser uses ALLOT of memory then it  parses
the huge jsonfile with distances.

We could solve this by running as 64 bit  or better but more work to  parse
distances in smaller parts.

2016-01-11 6:19 GMT+01:00 Aethir <notifications@github.com>:

**Comment by AnthorNet** [comment]
Yep even our API starts to take a lot of memory, I optimise a little for the new server on the systems call but the distances is more problematic, and even with those optimisations, the problem will come back once they are more distances.
It's a never ending problem.

What I could suggest is that we block the distances API as a whole and only accept queries when a system is set.
You would have to take distances on the fly, when you needing them for a particular system, but the query should be fast to process.

Actually the distances API, does not have any kind of selection by system name, I think that worked when we had 50K distances, but now it works less well ^^

We will soon provide nightly dump too, the process is running, I just have to write them.

**Comment by Aethir** [comment]
Awesome... that stopped the error.
Still no luck with getting to my travelhistory, though... 

it says it's loading everything successfully and then... that's it. 

I took a look at my netlogs and they're a near-continuous stream of what look like network routing info but nothing I could see that remotely pertains to system information...

I zipped both my AppConfig.xml and my latest netlog file and have posted here; if it's not too much to ask, could someone have a look at them and help me figure out what's going on? 

All help greatly appreciated, I think this is a fantastic program and I *really* want it to work!

[AppConfig.zip](https://github.com/EDDiscovery/EDDiscovery/files/88480/AppConfig.zip)
